classes/invitation/invitations/BaseInvitation.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @file classes/invitation/invitations/BaseInvitation.php
+ *
+ * Copyright (c) 2014-2023 Simon Fraser University
+ * Copyright (c) 2000-2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class BaseInvitation
+ *
+ *
+ * @brief Abstract class for all Invitations
+ */
+
+namespace PKP\invitation\invitations;
+
+use APP\core\Application;
+use Carbon\Carbon;
+use DateTime;
+use Exception;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Mail\Mailable;
+use phpDocumentor\Reflection\Types\Void_;
+use PKP\config\Config;
+use PKP\context\Context;
+use PKP\facades\Repo;
+use PKP\invitation\invitations\enums\InvitationStatus;
+use PKP\invitation\models\Invitation;
+use PKP\pages\invitation\PKPInvitationHandler;
+use PKP\security\Validation;
+use Symfony\Component\Mailer\Exception\TransportException;
+
+abstract class BaseInvitation
+{
+    use SerializesModels;
+
+    const DEFAULT_EXPIRY_DAYS = 3;
+
+    /**
+     * The name of the class name of the specific invitation
+     */
+    public string $className;
+    private string $keyHash;
+    public string $key;
+    public DateTime $expirationDate;
```

Comment:
The key, expiration date, and class name should be characteristics of the Eloquent model, not this class, I think.

Reply:
For now I am keeping the BaseInvitation as the Business class, so, I am keeping them here (also)